client-network/src/network_ops/default_network_ops_client.rs

```diff
@@ -472,8 +445,52 @@ where
     }
 
     #[inline]
-    fn get_staked_state(&self, address: &StakedStateAddress) -> Result<StakedState> {
-        self.get_staked_state_account(address)
+    fn get_staked_state(
+        &self,
+        name: &str,
+        address: &StakedStateAddress,
+        verify: bool,
+    ) -> Result<StakedState> {
+        let mstaking = if verify {
+            let sync_state = self.wallet_client.get_sync_state(name)?;
+            let rsp = self.client.query(
+                "staking",
+                address.as_ref(),
+                Some(sync_state.last_block_height.into()),
+                true,
+            )?;
+            let mstaking = <Option<StakedState>>::decode(&mut rsp.bytes().as_slice())
+                .err_kind(ErrorKind::DeserializationError, || {
+                    format!("Cannot deserialize staked state for address: {}", address)
+                })?;
+
+            let mut proof_bytes = rsp.proof.as_ref().unwrap().ops[0].data.as_slice();
```

Comment:
expect and
assert!(ops.len()>0)?

Reply:
Changed to return proper error for invalid response.